hrms/overrides/employee_advance.py

```diff
@@ -0,0 +1,201 @@
+            frappe.throw(_("Only the Added approver can approve"))
+
+        employee_advance_doc.docstatus = 1
+        employee_advance_doc.status = "Pending Approval By Admin"
+        CustomTask.mark_tasks_as_completed(
+            "Employee Advance", doc.name, TaskType.APPROVE_EMPLOYEE_ADVANCE
+        )
+    elif employee_advance_doc.status == "Pending Approval By Admin":
+        if not user_has_role(frappe.session.user, RoleEnum.FINANCE_ROLE.value):
+            frappe.throw(_("Only user having the role of Expense Admin can approve the document"))
+        employee_advance_doc.status = "Pending Approval By HR"
+    elif employee_advance_doc.status == "Pending Approval By HR":
+        if not user_has_role(frappe.session.user, RoleEnum.HR_ROLE.value):
+            frappe.throw(_("Only user having HR User can approve the document"))
+        employee_advance_doc.status = "Pending Payment"
+    employee_advance_doc.db_update()
+
+
+@frappe.whitelist()
+def create_payment_entry(doc, values):
+    import json
+
+    if isinstance(doc, str):
+        doc = frappe._dict(json.loads(doc))
+    if isinstance(values, str):
+        payment_values = frappe._dict(json.loads(values))
+
+    bank_cash_account = get_bank_cash_account(payment_values.mode_of_payment, doc.company).get(
+        "account"
+    )
+    bank_account_currency = get_account_details(bank_cash_account, frappe.utils.nowdate()).get(
+        "account_currency"
+    )
+    payment_entry = frappe.get_doc(
+        {
+            "doctype": "Payment Entry",
+            "docstatus": 1,
+            "payment_type": "Pay",
+            "mode_of_payment": payment_values.mode_of_payment,
+            "party_type": "Employee",
+            "party": doc.employee,
+            "party_name": doc.employee_name,
+            "paid_from": bank_cash_account,
+            "paid_from_account_currency": bank_account_currency,
+            "paid_to": "Employee Advances - Suite42",
+            "paid_to_account_currency": "INR",
+            "paid_amount": doc.sanctioned_amount,
+            "total_allocated_amount": doc.sanctioned_amount,
+            "received_amount": doc.sanctioned_amount,
+            "unallocated_amount": 0,
+            "references": [
+                {
+                    "reference_doctype": "Employee Advance",
+                    "reference_name": doc.name,
+                    "total_amount": doc.sanctioned_amount,
+                    "outstanding_amount": doc.sanctioned_amount,
+                    "allocated_amount": doc.sanctioned_amount,
+                    "parentfield": "references",
+                    "parenttype": "Payment Entry",
+                    "docstatus": 1,
+                    "doctype": "Payment Entry Reference",
+                }
+            ],
+            "reference_no": payment_values.reference_no,
+            "reference_date": payment_values.payment_dates,
```

Comment:
rename payment_values.payment_dates to payment_values.payment_date

Reply:
Done